dsls/sql/src/main/java/org/apache/beam/dsls/sql/schema/BeamSqlRowCoder.java

```diff
@@ -67,24 +66,24 @@ public void encode(BeamSQLRow value, OutputStream outStream,
 
       switch (value.getDataType().getFieldsType().get(idx)) {
       case INTEGER:
+        intCoder.encode(value.getInteger(idx), outStream, nested);
+        break;
       case SMALLINT:
       case TINYINT:
-        intCoder.encode(value.getInteger(idx), outStream, nested);
+        intCoder.encode((int) value.getShort(idx), outStream, nested);
         break;
       case DOUBLE:
-      case FLOAT:
         doubleCoder.encode(value.getDouble(idx), outStream, nested);
         break;
+      case FLOAT:
+        doubleCoder.encode((double) value.getFloat(idx), outStream, nested);
+        break;
       case BIGINT:
         longCoder.encode(value.getLong(idx), outStream, nested);
         break;
       case VARCHAR:
         stringCoder.encode(value.getString(idx), outStream, nested);
         break;
-      case TIME:
```

Comment:
It's moved to a separated task, as a hard dependency of _window support_ in task 2006.